doc/changelog.rst

```diff
@@ -38,6 +49,11 @@ Unavoidable breaking changes
   Users of the Stable API with estimated_document_count are recommended to upgrade
   their server version to 5.0.9+ or set :attr:`pymongo.server_api.ServerApi.strict`
   to ``False`` to avoid encountering errors (`PYTHON-3167`_).
+- Removed generic typing from :class:`~pymongo.client_session.ClientSession` to improve
+  support for pyright (`PYTHON-3283`_).
```

Comment:
Another nitpick, but it seems according to https://github.com/Microsoft/pyright the correct capitalization is `Pyright`.